service/pkg/cache/cache.go

```diff
@@ -0,0 +1,115 @@
+package cache
+
+import (
+	"context"
+	"log"
+	"time"
+
+	"github.com/dgraph-io/ristretto"
+	"github.com/eko/gocache/lib/v4/cache"
+	"github.com/eko/gocache/lib/v4/store"
+	ristretto_store "github.com/eko/gocache/store/ristretto/v4"
+)
+
+type Manager struct {
+	cache *cache.Cache[interface{}]
+}
+
+// Cache is a cache implementation using gocache
+type Cache struct {
+	manager      *Manager
+	serviceName  string
+	cacheOptions Options
+}
+
+type Options struct {
+	Expiration time.Duration
+	Cost       int64
+}
+
+// NewCache creates a new Cache instance using Ristretto as the backend.
+func NewCacheManager(maxCost int64) (*Manager, error) {
+	numCounters, bufferItems := EstimateRistrettoConfigParams(maxCost)
+	store, err := ristretto.NewCache(&ristretto.Config{
+		NumCounters: numCounters, // number of keys to track frequency of (10x max items)
+		MaxCost:     maxCost,     // maximum cost of cache (e.g., 1<<20 for 1MB)
+		BufferItems: bufferItems, // number of keys per Get buffer.
+	})
+	if err != nil {
+		return nil, err
+	}
+	ristrettoStore := ristretto_store.NewRistretto(store)
+	c := cache.New[interface{}](ristrettoStore)
+	return &Manager{
+		cache: c,
+	}, nil
+}
+
+// NewCache creates a new Cache instance with the given service name and options.
+// The purpose of this function is to create a new cache for a specific service.
+// Because caching can be expensive we want to make sure there are some strict controls with
+// how it is used.
+func (c *Manager) NewCache(serviceName string, options Options) *Cache {
```

Comment:
Do we want to pass in the logger of the service OR have a cache manage logger. 

@jakedoublev brought up the with multiple versions of Authorization will need to pass in the logger otherwise we wont have access to the specific log attribute version